apps/back/src/Entity/Expense.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace App\Entity;
+
+use DateTime;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\ORM\Mapping\Column;
+use Doctrine\ORM\Mapping\Entity;
+use Doctrine\ORM\Mapping\GeneratedValue;
+use Doctrine\ORM\Mapping\Id;
+use Doctrine\ORM\Mapping\JoinColumn;
+use Doctrine\ORM\Mapping\JoinTable;
+use Doctrine\ORM\Mapping\ManyToMany;
+use Doctrine\ORM\Mapping\ManyToOne;
+use Doctrine\ORM\Mapping\Table;
+
+#[Entity]
+#[Table(name: 'expense')]
+class Expense {
+    #[Id]
+    #[Column(type: Types::INTEGER)]
+    #[GeneratedValue(strategy:'IDENTITY')]
+    private int $id;
+
+    #[Column(length: 140)]
+    private string $name;
+
+    #[Column(type: Types::DATETIME_IMMUTABLE)]
+    private \DateTimeImmutable $date;
+
+    #[Column(type: Types::INTEGER)]
+    private int $amount;
+
+    #[ManyToMany(targetEntity: User::class, inversedBy:'expensesAsParticipant')]
+    #[JoinTable(name: 'users_expenses_as_participant')]
+    private Collection $participants;
+
+    #[ManyToOne(targetEntity: User::class, inversedBy: 'expensesAsPayer')]
+    #[JoinColumn(name: 'payer_id', referencedColumnName: 'id')]
+    private User $payer;
+
+    #[ManyToOne(targetEntity: Group::class, inversedBy: 'expensesOfTheGroup')]
+    #[JoinColumn(name: 'group_id', referencedColumnName: 'id')]
+    private Group $group;
+
+    public function __construct(
+        string $name,
+        \DateTimeImmutable $date,
+        int $amount,
+        array $participants,
+        User $payer,
+        Group $group
+    ) {
+        $this->name = $name;
+        $this->date = $date;
+        $this->amount = $amount;
+        $this->participants = $participants;
+        $this->payer = $payer;
+        $this->group = $group;
+    }
```

Comment:
Comme pour ton modèle, tu dois pouvoir initialiser tes propriétés de cette manière pour éviter la redondance.

```suggestion
    public function __construct(
        #[Id]
        #[Column(type: Types::INTEGER)]
        #[GeneratedValue(strategy:'IDENTITY')]
        private int $id;

        #[Column(length: 140)]
        private string $name;

        #[Column(type: Types::DATETIME_IMMUTABLE)]
        private \DateTimeImmutable $date;

        #[Column(type: Types::INTEGER)]
        private int $amount;

        #[ManyToMany(targetEntity: User::class, inversedBy:'expensesAsParticipant')]
        #[JoinTable(name: 'users_expenses_as_participant')]
        private Collection $participants;

        #[ManyToOne(targetEntity: User::class, inversedBy: 'expensesAsPayer')]
        #[JoinColumn(name: 'payer_id', referencedColumnName: 'id')]
        private User $payer;

        #[ManyToOne(targetEntity: Group::class, inversedBy: 'expensesOfTheGroup')]
        #[JoinColumn(name: 'group_id', referencedColumnName: 'id')]
        private Group $group;
    ) {
    }
```